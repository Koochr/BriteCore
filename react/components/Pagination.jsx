import React from "react"
import PropTypes from "prop-types"

const Pagination = props => {
	const firstDisplayedNumber = (props.currentPage * props.rowsPerPage) + 1
	const lastDisplayedNumber = Math.min((props.currentPage + 1) * props.rowsPerPage, props.totalRows)

	return <ul className="pagination">
		{props.currentPage !== 0 && <li
			className="page-item page-link"
			onClick={() => {props.onChangePage(props.currentPage-1)}}
		>
      Previous
		</li>}
		<li className="page-item navbar-brand pagination-text">
			{` Records ${firstDisplayedNumber} - ${lastDisplayedNumber} of ${props.totalRows} `}
		</li>
		{props.currentPage < props.totalRows/props.rowsPerPage - 1 && <li
			className="page-item page-link"
			onClick={() => {props.onChangePage(props.currentPage+1)}}
		>
      Next
		</li>}
	</ul>
}

Pagination.propTypes = {
	currentPage: PropTypes.number.isRequired,
	rowsPerPage: PropTypes.number.isRequired,
	totalRows: PropTypes.number.isRequired,
	onChangePage: PropTypes.func.isRequired,
}

export default Pagination
