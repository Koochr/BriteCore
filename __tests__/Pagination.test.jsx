/* global it, jest, expect */
import React from "react"
import { shallow, configure, mount, } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import Pagination from "../react/components/Pagination.jsx"

configure({ adapter: new Adapter(), })

it("Should render correctly", () => {
	const component = shallow(
		<Pagination onChangePage={() => {}} currentPage={0} totalRows={200} rowsPerPage={15}/>
	)
	expect(component).toMatchSnapshot()
})

it("Should render needed elements", () => {
	const component = shallow(
		<Pagination onChangePage={() => {}} currentPage={2} totalRows={200} rowsPerPage={15}/>
	)
	const ul = component.find("ul")
	expect(ul).toHaveLength(1)
	const buttons = component.find(".page-link")
	expect(buttons).toHaveLength(2)
	const text = component.find(".pagination-text")
	expect(text).toHaveLength(1)
	expect(text.props().children).toStrictEqual(" Records 31 - 45 of 200 ")
})

it("Should handle change page", () => {
	const mockOnChangePage = jest.fn()
	const component = mount(
		<Pagination onChangePage={mockOnChangePage} currentPage={2} totalRows={200} rowsPerPage={15}/>
	)
	const buttons = component.find(".page-link")
	buttons.at(0).simulate("click")
	expect(mockOnChangePage).toBeCalledWith(1)
	buttons.at(1).simulate("click")
	expect(mockOnChangePage).toBeCalledWith(3)
})

