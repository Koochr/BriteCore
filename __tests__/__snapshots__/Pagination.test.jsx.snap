// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    currentPage={0}
    onChangePage={[Function]}
    rowsPerPage={15}
    totalRows={200}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <li
          className="page-item navbar-brand pagination-text"
        >
           Records 1 - 15 of 200 
        </li>,
        <li
          className="page-item page-link"
          onClick={[Function]}
        >
          Next
        </li>,
      ],
      "className": "pagination",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Records 1 - 15 of 200 ",
          "className": "page-item navbar-brand pagination-text",
        },
        "ref": null,
        "rendered": " Records 1 - 15 of 200 ",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Next",
          "className": "page-item page-link",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Next",
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <li
            className="page-item navbar-brand pagination-text"
          >
             Records 1 - 15 of 200 
          </li>,
          <li
            className="page-item page-link"
            onClick={[Function]}
          >
            Next
          </li>,
        ],
        "className": "pagination",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Records 1 - 15 of 200 ",
            "className": "page-item navbar-brand pagination-text",
          },
          "ref": null,
          "rendered": " Records 1 - 15 of 200 ",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Next",
            "className": "page-item page-link",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Next",
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
